azblob/highlevel.go

```diff
@@ -208,7 +208,7 @@ func downloadBlobToBuffer(ctx context.Context, blobURL BlobURL, offset int64, co
 			if err != nil {
 				return err
 			}
-			body := dr.Body(o.RetryReaderOptionsPerBlock)
+			body := dr.Body(o.RetryReaderOptionsPerBlock, nil)
```

Comment:
highlevel.go is just a highlevel example library and we are not equipping it to work with encryption. That's why we are just passing nil here. ObjectStore use direct API and not highlevel.go